openlibrary/plugins/upstream/mybooks.py

```diff
@@ -342,8 +343,20 @@ def GET(self, username, key='want-to-read'):
                 }
                 for w in books
             ]
+
+            if len(records_json) < limit:
+                num_found = len(records_json)
+            else:
+                num_found = readlog.count_shelf(key)
```

Comment:
Ah bug! I was trying to avoid fetching the proper count when it was unneeded.

```suggestion
            if page ==1 and len(records_json) < limit:
                num_found = len(records_json)
            else:
                num_found = readlog.count_shelf(key)
```